lib/open_food_network/scope_variant_to_hub.rb

```diff
@@ -29,21 +29,15 @@ def in_stock?
       end
 
       def count_on_hand
-        @variant_override.andand.count_on_hand || super
+        return super unless @variant_override.andand.stock_overridden?
```

Comment:
It's been a while I don't read VOs code but `@variant_override` is set at this point, right? then we don't need to `andand` defensive programming. The same happens with all the other methods using it.

Reply:
no, @variant_override is nil for variants that are not overriden (this code will be executed for all calls to variant.count_on_hand :-)
I think the alternative is something like:
```
return super unless @variant_override.present? && @variant_override.stock_overridden?
```
andand is shorter.

Reply:
I was wondering why you didn't use `andand` in the next line and it took me half a minute to make sure that this logic is correct. I would have found this version easier to read:
```rb
if @variant_override.andand.stock_overridden?
  @variant_override.count_on_hand
else
  super
end
```
A bit more rubyish, more functional, less procedural. But maybe that's just my preference?

Reply:
I ended up making count_on_hand and on_demand be a simple if else statement and without using andand. see new commit.